### DISCUSSION TITLE
Usage of Profile Custom Attributes (Exchange custom attributes 1-15) in the PNP Search as filter

### DISCUSSION THREAD

**Discussion by msteuper** [microsoft-search/pnp-modern-search#4112] [Q&A]
Hi all, 
we have extended the profile card with values out of the Exchange custom attribute set in Entra ID which are filled via a sync from the local AD. As mentioned, the values are displayed in the profile card, but we would also like to provide a people search in the intranet which should also provide the option to filter via these custom attributes in the PNP search context. Any idea how this can be achieved? Thanks a lot for any hints.

**Comment by wobba** [comment]
Unfortunately the only "good" people search API uses SharePoint search API, which means you need to sync those fields over to UPA properties which you can then use in search. 

See https://learn.microsoft.com/en-us/sharepoint/dev/solution-guidance/bulk-user-profile-update-api-for-sharepoint-online#updating-the-sharepoint-online-user-profile-using-pnp-powershell and also see the script at https://pnp.github.io/script-samples/spo-request-pnp-reindex-user-profile/README.html for reference.

**Reply by DavorSukeri** [reply]
@wobba I tried to trigger user profile reindexing with some older script (by you, I think) with the SPS-Birthday field, but for some reason it just skipped all the users because they did not have a value in the SPS-Birthday field. I assume it was supposed to add current date value to it if it was empty and then write it back to empty.

I have done this before and I think it was the same older script, but at that time I changed the Department-field value. That worked fine except that I noticed the values it wrote back to the profiles has some encoding issues. For example all the nordic letters ÄÖ were encoded.

Can I use the script you linked, but use with some other field than Department? Maybe the SPS-Birthday field or some other field that does not have a value at all so there's no risk with encoding issues?

**Reply by DavorSukeri** [reply]
nvm, I just fired up this https://pnp.github.io/script-samples/spo-request-pnp-reindex-user-profile/README.html?tabs=pnpps and seems to be working.

**Comment by jwiersem** [comment]
Any update for the MS Search part of PnP modern search @wobba? The SharePoint profile is not futureproof anymore, i think you cant even edit it anymore as a user since Delve is gone, correct? When does MS Search for profiles reach feature parity with SharePoint search and SP profiles?
